extensions/sql-database-projects/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -88,16 +88,68 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined,
 	let connectionStringSettingName;
 
 	// show the settings from project's local.settings.json if there's an AF functions project
-	// TODO: allow new setting name to get added here and added to local.settings.json
 	if (project) {
-		const settings = await azureFunctionsUtils.getLocalSettingsJson(path.join(path.dirname(project!), constants.azureFunctionLocalSettingsFileName));
-		const existingSettings: string[] = settings.Values ? Object.keys(settings.Values) : [];
+		let settings;
+		try {
+			settings = await azureFunctionsUtils.getLocalSettingsJson(path.join(path.dirname(project!), constants.azureFunctionLocalSettingsFileName));
+		} catch (e) {
+			void vscode.window.showErrorMessage(e);
+			return;
+		}
+
+		let existingSettings = settings?.Values ? Object.keys(settings.Values).map(setting => {
+			return {
+				label: setting
+			} as vscode.QuickPickItem & { isCreateNew?: boolean };
+		}) : [];
+		existingSettings.unshift({ label: constants.createNewLocalAppSetting, isCreateNew: true });
+
+		while (!connectionStringSettingName) {
+			const selectedSetting = await vscode.window.showQuickPick(existingSettings, {
+				canPickMany: false,
+				title: constants.selectSetting,
+				ignoreFocusOut: true
+			});
+			if (!selectedSetting) {
+				// User cancelled
+				return;
+			}
+
+			if (selectedSetting.isCreateNew) {
+				const newConnectionStringSettingName = await vscode.window.showInputBox(
+					{
+						title: constants.enterConnectionStringSettingName,
+						ignoreFocusOut: true,
+						validateInput: input => input ? undefined : constants.nameMustNotBeEmpty
+					}
+				) ?? '';
+
+				if (newConnectionStringSettingName) {
+					const newConnectionStringValue = await vscode.window.showInputBox(
+						{
+							title: constants.enterConnectionString,
+							ignoreFocusOut: true,
+							validateInput: input => input ? undefined : constants.valueMustNotBeEmpty
+						}
+					) ?? '';
+
+					if (!newConnectionStringValue) {
+						// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
+						continue;
+					}
+
+					const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(project), newConnectionStringSettingName, newConnectionStringValue);
```

Comment:
This can throw in a variety of ways - make sure you're handling those cases (possibly just by displaying message and showing the quickpick again)